### DISCUSSION TITLE
Which rule corresponds to which errors?

### DISCUSSION THREAD

**Discussion by Meemaw** [microsoft/pyright#4146] [General]
We are seeing the following errors in VSCode (reported by Pylance).

`Argument of type "Account | None" cannot be assigned to parameter "account" of type "Account" `. This happens when we try to call a function with `Account | None` that takes just `Account`.

This error is totally valid, however running `pyright` (with our custom config) does not report this error.

We don't have all the rules enabled by default, so it would be great to know which rule would catch that.

I looked at the [configuration options](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#type-check-diagnostics-settings), but could not find a match.

**Comment by erictraut** [comment]
Pylance uses pyright under the covers, so you will get the same behaviors if you configure them the same. There are three common reasons for seeing differences between pylance and pyright:

1. Pylance bundles additional stubs that pyright does not. These stubs are located in [this repo](https://github.com/microsoft/python-type-stubs), so you can clone the applicable stubs and check them in to your repo (in the "typings" subdirectory, typically).
2. Pylance defaults `useLibraryCodeForTypes` to true, whereas pyright defaults to false. This configuration setting controls what pyright does when it encounters a library with no "py.typed" file and no type stubs. If `useLibraryCodeForTypes` is enabled, pyright will attempt to infer type information from the available library code. The resulting types are usually sufficient for completion suggestions and other language server features, but they're often not accurate enough for type checking. If you enable type checking, then we recommend leaving `useLibraryCodeForTyeps` off.
3. Pylance and pyright are both released weekly, but pylance normally lags pyright by one week (unless you use the insiders build of pylance, in which case they are in sync). This means that there could be temporary behavior differences due to minor bug fixes, etc. This is rare though.
